src/tinystr16.rs

```diff
@@ -223,6 +228,33 @@ impl TinyStr16 {
                 >> 2);
         unsafe { Self(NonZeroU128::new_unchecked(result)) }
     }
+
+    #[inline(always)]
+    fn try_from_slice<S: AsRef<[u8]>>(text: S) -> Result<Self, Error> {
```

Comment:
Just make this take `&[u8]` directly, and use `as_bytes()` in the `&str` caller? Seems simpler.